hack/generator/pkg/astmodel/armconversion/shared.go

```diff
@@ -40,6 +41,30 @@ func (builder conversionBuilder) propertyConversionHandler(
 	panic(fmt.Sprintf("No property found for %s", toProp.PropertyName()))
 }
 
+// deepCopyJSON special cases copying JSON-type fields to call the DeepCopy method.
+// It generates code that looks like:
+//     <destination> = *<source>.DeepCopy()
+func (builder *conversionBuilder) deepCopyJSON(
+	params complexPropertyConversionParameters) []ast.Stmt {
+	newSource := &ast.UnaryExpr{
+		X: &ast.CallExpr{
+			Fun: &ast.SelectorExpr{
+				X:   params.source,
+				Sel: ast.NewIdent("DeepCopy"),
+			},
+			Args: []ast.Expr{},
+		},
+		Op: token.MUL,
+	}
+	assignmentHandler := params.assignmentHandler
+	if assignmentHandler == nil {
+		assignmentHandler = assignmentHandlerAssign
+	}
+	return []ast.Stmt{
+		assignmentHandler(params.destination, newSource),
+	}
+}
+
```

Comment:
Once my Test Cases branch is merged, we should loop back here and use some of the helper methods I've written to simplify the code.